### PR TITLE
HttpUrlCondition extracted

### PR DESCRIPTION
**Pull request by koral--**

### CODE REVIEW

File: dispatcher/src/main/kotlin/pl/droidsonroids/testing/mockwebserver/condition/Condition.kt [hunk 1]
```diff
@@ -1,15 +1,16 @@
 package pl.droidsonroids.testing.mockwebserver.condition
 
-import okhttp3.HttpUrl
+import okhttp3.mockwebserver.RecordedRequest
 
 /**
  * Represents set of mocked request properties which have to match received request.
  * Conditions are checked by <code>FixtureDispatcher</code> in order defined by values returned from <code>compareTo</code>.
```
Comment (FaultLessDroid) [inline dispatcher/src/main/kotlin/pl/droidsonroids/testing/mockwebserver/condition/Condition.kt]: INFO Line detected that is longer than the defined maximum line length in the code style. [rule](http://sonar.droidsonroids.pl/coding_rules#rule_key=detekt-kotlin%3AMaxLineLength)

**Comment by FaultLessDroid** [comment]
SonarQube analysis reported 2 issues
* ![INFO][INFO] 2 info

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [PathQueryCondition.kt#L7](https://github.com/DroidsOnRoids/mockwebserver-path-dispatcher/blob/a55f38f337a2e471231e4c6bd7a6ff07589bd0c2/dispatcher/src/main/kotlin/pl/droidsonroids/testing/mockwebserver/condition/PathQueryCondition.kt#L7): Line detected that is longer than the defined maximum line length in the code style. [rule](http://sonar.droidsonroids.pl/coding_rules#rule_key=detekt-kotlin%3AMaxLineLength)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

File: dispatcher/src/main/kotlin/pl/droidsonroids/testing/mockwebserver/condition/HttpUrlCondition.kt [hunk 2]
```diff
@@ -0,0 +1,23 @@
+package pl.droidsonroids.testing.mockwebserver.condition
+
+import okhttp3.HttpUrl
+import okhttp3.mockwebserver.RecordedRequest
+
+/**
+ * A Condition which matches HttpUrls only, they are extracted from request so that only
+ * <code>isUrlMatching(HttpUrl)</code> needs to implemented.
```
Comment (burnoo) [inline dispatcher/src/main/kotlin/pl/droidsonroids/testing/mockwebserver/condition/HttpUrlCondition.kt]: bardziej czytelniejsze jest chyba `{@code ...}` od `<code>...</code>`
https://stackoverflow.com/questions/7066409/what-is-the-difference-between-code-memberdata-and-codememberdata-code-in

Comment (koral--) [inline dispatcher/src/main/kotlin/pl/droidsonroids/testing/mockwebserver/condition/HttpUrlCondition.kt]: to jest kdoc, nie javadoc, tu nie ma @code ;)
https://kotlinlang.org/docs/reference/kotlin-doc.html
ale można użyć `[]` sprawdzę tę opcję

Comment (burnoo) [inline dispatcher/src/main/kotlin/pl/droidsonroids/testing/mockwebserver/condition/HttpUrlCondition.kt]: ok

**Pull request merged**